### ISSUE TITLE
Does threddsIso handle the 'source' CF attribute?

### ISSUE THREAD

**Issue by mwengren** [Unidata/threddsIso#34]
See: http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html#description-of-file-contents.

I did a search in the current version of [UnidataDD2MI.xsl](https://github.com/Unidata/threddsIso/blob/master/src/main/resources/xsl/nciso/UnidataDD2MI.xsl) and couldn't find it called out in the variables section at the top.  

Since this is mentioned as a critical file description variable in CF, shouldn't it be parsed and output in the ISO XML somewhere?

**Comment by lukecampbell** [comment]
As of ACDD 1.1 there was no mapping or use of the global `source` attribute.

http://wiki.esipfed.org/index.php/Attribute_Convention_for_Data_Discovery_1-1

`source` is defined in ACDD 1.3 but I don't believe a mapping has been done from ACDD 1.3 to ISO-19115-2 to date.

**Comment by mwengren** [comment]
Hey @lukecampbell.  I read down to this [comment](https://github.com/Unidata/threddsIso/issues/7#issuecomment-293731847) in issue #7 noticed that is is included in the new mapping spreadsheet.  

I don't know how long `source` has been part of CF, but since it's alongside the other high level identification variables I would guess awhile.  Might have just been overlooked in earlier ncISO versions.  

Hopefully, it will be included in the upcoming release!  (this will help us tremendously at IOOS to identify model datasets by some automated fashion for potential inclusion in EDS via the [Catalog](https://data.ioos.us), at least, if CKAN indexes where it ends up in the ISO XML for search/filter, a separate issue potentially).  See: https://github.com/ioos/catalog/issues/40

**Comment by ocefpaf** [comment]
Just to xref. another issue here: this would really help with https://github.com/ioos/catalog/issues/40

**Comment by lesserwhirls** [comment]
Tagging @noaaroland - do the transforms in https://github.com/NOAA-PMEL/uafnciso handle the source attribute?

**Comment by noaaroland** [comment]
It does now: https://github.com/NOAA-PMEL/uafnciso/releases/tag/2.3.6

**Comment by noaaroland** [comment]
The CDN versions of the transforms I use in the code are:
https://cdn.jsdelivr.net/gh/NOAA-PMEL/uafnciso@a9f3dceb0dbc092f8d7f5153ceeb58af2d30edc4/transforms/UnidataDD2MI.xsl
and
https://cdn.jsdelivr.net/gh/NOAA-PMEL/uafnciso@fdb7f86515c21a8b5c087978975addf9ad5d0027/transforms/UnidataDDCount-HTML.xsl

**Comment by mwengren** [comment]
@noaaroland Thanks for updating the NOAA-PMEL/uafnciso repo with fixes! 

Since this is a fork of the original ncISO, I presume it won't be included in any new TDS releases, correct?  

We can point our data providers to use the uafnciso release JAR files as an alternative to generate ISO files from TDS.   If, that is, we are able to leverage that for filtering capability based on the values populated in 'source' - one of the original goals of ioos/catalog#40.

Not sure if this should be closed or left open until this functionality is ported here if that's desired.  Feel free to close if not.

**Comment by lesserwhirls** [comment]
I think we are still in a place where there have been changes made to the transforms in this repository, and there have been changes made to the transforms in the NOAA-PMEL repository, and they have diverged enough to where I am unable to make a call on what stays and what goes (plus, I'm not exactly proficient with xslt stuff, so you probably don't want me doing it anyways). It'd be great if we could settle around a canonical set, and I'd be in favor of adopting them here (not my call, however).

**Comment by dneufeldcu** [comment]
I agree with consolidating the changes in the Unidata threddisIso repo, I think that was the intent behind merging the client code into that repository as well. Not sure who is best positioned to review and merge the XSLT at this point? @amilan17 @noaaroland

**Comment by zdefne-usgs** [comment]
We have also made some changes to XSLT to address some of the issues such as:
https://github.com/Unidata/threddsIso/issues/41
The most recent copy can be found at:
https://github.com/zdefne-usgs/ocean-iso-metadata/tree/master/XSL/usgs_current
